sdv/metadata/single_table.py

```diff
@@ -1030,8 +1031,15 @@ def _get_invalid_column_values(column, validation_function):
 
         return set(column[~valid])
 
-    def _validate_column_data(self, column):
-        """Validate values of the column satisfy its sdtype properties."""
+    def _validate_column_data(self, column, sdtype_warnings):
+        """Validate values of the column satisfy its sdtype properties.
```

Comment:
can we update the return type of this method? I think it's a little confusing because usually our validate methods raise errors but here it returns list with the error message. Similarly, we are appending warnings to a list instead of raising them and bubbling them up. This is fine but I think a description could help